graylog2-server/src/test/java/org/graylog2/security/ldap/LdapConnectorTest.java

```diff
@@ -0,0 +1,256 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.graylog2.security.ldap;
+
+import org.apache.directory.ldap.client.api.LdapConnectionConfig;
+import org.apache.directory.ldap.client.api.LdapNetworkConnection;
+import org.apache.directory.server.annotations.CreateLdapServer;
+import org.apache.directory.server.annotations.CreateTransport;
+import org.apache.directory.server.core.annotations.ApplyLdifs;
+import org.apache.directory.server.core.annotations.ContextEntry;
+import org.apache.directory.server.core.annotations.CreateDS;
+import org.apache.directory.server.core.annotations.CreateIndex;
+import org.apache.directory.server.core.annotations.CreatePartition;
+import org.apache.directory.server.core.annotations.LoadSchema;
+import org.apache.directory.server.core.integ.AbstractLdapTestUnit;
+import org.apache.directory.server.core.integ.FrameworkRunner;
+import org.apache.directory.server.core.partition.impl.avl.AvlPartition;
+import org.apache.directory.server.ldap.LdapServer;
+import org.graylog2.shared.security.ldap.LdapEntry;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(FrameworkRunner.class)
+@CreateLdapServer(transports = {
+        @CreateTransport(protocol = "LDAP")
+})
+@CreateDS(
+        name = "LdapConnectorTest",
+        partitions = {
+                @CreatePartition(
+                        name = "example.com",
+                        type = AvlPartition.class,
+                        suffix = "dc=example,dc=com",
+                        contextEntry = @ContextEntry(
+                                entryLdif = "dn: dc=example,dc=com\n" +
+                                            "dc: example\n" +
+                                            "objectClass: top\n" +
+                                            "objectClass: domain\n\n"
+
+                        ),
+                        indexes = {
+                                @CreateIndex(attribute = "objectClass"),
+                                @CreateIndex(attribute = "dc"),
+                                @CreateIndex(attribute = "ou")
+                        }
+
+                )
+        },
+        loadedSchemas = {
+                @LoadSchema(name = "nis", enabled = true)
+        }
+)
+@ApplyLdifs(
```

Comment:
Could we replace this with `@ApplyLdifFiles` so that the LDIF could also be easily applied to another LDAP server?